hudi-client/hudi-spark-client/src/main/java/org/apache/hudi/common/model/HoodieSparkRecord.java

```diff
@@ -307,7 +309,12 @@ public Option<Map<String, String>> getMetadata() {
 
   @Override
   public Option<HoodieAvroIndexedRecord> toIndexedRecord(Schema recordSchema, Properties prop) {
-    throw new UnsupportedOperationException();
+    if (data == null) {
+      return Option.empty();
+    }
+    StructType structType = schema == null ? AvroConversionUtils.convertAvroSchemaToStructType(recordSchema) : schema;
+    GenericRecord convertedRecord = AvroConversionUtils.createInternalRowToAvroConverter(structType, recordSchema, false).apply(data);
+    return Option.of(new HoodieAvroIndexedRecord(key, convertedRecord));
```

Comment:
Where is this method invoked now that would throw `UnsupportedOperationException` without these changes?

Reply:
In the index utils there is a merge step that compares the values in `mergeIncomingWithExistingRecord`